packages/salesforcedx-utils-vscode/src/telemetry/telemetry.ts

```diff
@@ -233,4 +266,38 @@ export class TelemetryService {
     const hrend = process.hrtime(hrstart);
     return Number(util.format('%d%d', hrend[0], hrend[1] / 1000000));
   }
+
+  private getHasTelemetryMessageBeenShown(): boolean {
+    if (this.context === undefined) {
+      return true;
+    }
+
+    const sfdxTelemetryState = this.context.globalState.get(
+      TELEMETRY_GLOBAL_VALUE
+    );
+
+    return typeof sfdxTelemetryState === 'undefined';
+  }
+
+  private setTelemetryMessageShowed(): void {
+    if (this.context === undefined) {
+      return;
+    }
+
+    this.context.globalState.update(TELEMETRY_GLOBAL_VALUE, true);
+  }
+
+  /**
+   * Helper to run a callback if telemetry has been initialized and is
+   * enabled.
+   *
+   * @param callback function to call if telemetry is enabled
+   */
+  private validateTelemetry(callback: () => void): void {
+    if (this.reporter !== undefined) {
+      this.isTelemetryEnabled()
+        .then(enabled => Promise.resolve(enabled ? callback() : undefined))
+        .catch(err => console.error(err));
+    }
+  }
```

Comment:
In order to conform to the "no unhandled promise rejection" rule, i could either have A) updated the dozens of places in core that call telemetry methods with an empty `.catch()`, or B) change the telemetry methods to not return promises. I can't think of a reason why we'd want to wait for a telemetry method to resolve, so I decided to go with B and handle all the calls using this method. Let me know if you see an issue with doing it this way.

Reply:
This sounds good to me - you might need to update again bc of the changes I merged for the Apex package yesterday :/

Reply:
Don't we need to update the current places using the async methods (`await telemetryService.method`) using this approach?

Reply:
Using await shouldn't adversely affect it but it doesn't make sense to have it anymore, I'll update those places